public/locations-list-map.js

```diff
@@ -278,125 +377,103 @@ window.onFilterChange = function (elem, scrollNeeded) {
-  "VT": { "lat": 44.258803, "lng": -72.577841, zoom: 6.9 },
-  "WA": { "lat": 47.751074, "lng": -120.740139, zoom: 6.4 },
-  "WI": { "lat": 44.78444, "lng": -88.787868, zoom: 6.2 },
-  "WV": { "lat": 38.897626, "lng": -80.454903, zoom: 6.75 },
-  "WY": { "lat": 43.075968, "lng": -107.2902, zoom: 6.25 }
 };
 
-function initMap(states) {
-     var data_by_location = window.data_by_location;
-     var middle_of_us = { lat: 39.0567939, lng: -94.6065124};
-
-     var element = document.getElementById('map');
-
-     if (element == null) {
-         alert('could not find map div');
-     }
-
-    $(".map-container").show();
-
-     const singleStateFilter = states && states.length === 1;
-     const firstState = states[0];
-
-     // The map, roughly zoomed to show the entire US.
-     var map = new google.maps.Map( element, {zoom: 4, center: middle_of_us});
-
-     let markers = [];
-
-     // filter states if there is a state filter
-     const filteredStates = Object.keys(data_by_location).filter((stateCode) => (
-       (
-         singleStateFilter
-         && firstState.toUpperCase() === stateCode.toUpperCase()
-         && stateLocationMappings[stateCode.toUpperCase()]
-         // add markers when no single state filter or if state code is invalid
-       ) || (!singleStateFilter || !stateLocationMappings[firstState.toUpperCase()])
-     ));
-
-     for (const state of filteredStates.sort()) {
-         const cities = data_by_location[state];
-
-         for (const city of Object.keys(cities).sort()) {
-             for (const entry of cities[city]) {
-                 const name = entry["name"];
-                 const address = entry["address"];
-                 const latitude = entry["lat"];
-                 const longitude = entry["lng"];
-                 const instructions = entry["instructions"];
-                 const accepting = entry["accepting"];
-                 const open_accepted = entry["open_box"];
-                 // Convert the lat and lng fields to numbers
-                 if (!isNaN(Number(latitude))) {
-                     var marker = addMarkerToMap(map, Number(latitude), Number(longitude),
-                         address, name, instructions, accepting, open_accepted);
-                     markers.push(marker);
-                 }
-             }
-         }
-     }
-
-     if (singleStateFilter) {
-       // Center the map to a state if only one is set in query params
-       centerMapToState(map, firstState);
-     } else {
-       // Center the map on the nearest markers to the user if possible
-       centerMapToNearestMarkers(map, markers);
-     }
+
+
+/**
+ * Called once Google Maps API is fully loaded ready to run.
+ *
+ * IMPORTANT: This is exposed via window.initMap so the script can execute it.
+ */
+function initMap() {
+  var data_by_location = window.data_by_location;
+
+  var element = document.getElementById('map');
+
+  if (element == null) {
+    alert('could not find map div');
```

Comment:
Should not pop an alert message.

Reply:
Yeah, I abhor `alert()` in any real app. Not my introduced change (this was a refactor) but I can remove!